proxy/src/main/java/com/velocitypowered/proxy/tablist/VelocityTabList.java

```diff
@@ -0,0 +1,113 @@
+package com.velocitypowered.proxy.tablist;
+
+import com.google.common.base.Preconditions;
+import com.velocitypowered.api.proxy.player.TabList;
+import com.velocitypowered.api.proxy.player.TabListEntry;
+import com.velocitypowered.api.util.GameProfile;
+import com.velocitypowered.api.util.UuidUtils;
+import com.velocitypowered.proxy.connection.MinecraftConnection;
+import com.velocitypowered.proxy.protocol.packet.HeaderAndFooter;
+import com.velocitypowered.proxy.protocol.packet.PlayerListItem;
+import net.kyori.text.Component;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+
+public class VelocityTabList implements TabList {
+    private final MinecraftConnection connection;
+    private final Map<UUID, TabListEntry> entries = new HashMap<>();
```

Comment:
`ConcurrentHashMap` should be used here - there's the possibility of concurrent access.